core/src/main/kotlin/maru/consensus/ForkId.kt

```diff
@@ -0,0 +1,52 @@
+/*
+   Copyright 2025 Consensys Software Inc.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+ */
+package maru.consensus
+
+import maru.core.Hasher
+import maru.serialization.Serializer
+
+data class ForkId<T : ConsensusConfig>(
```

Comment:
Not sure why we need this data type and the serializer. I was thinking the forkId calculation would be quite simple with something like `forkId = Hasher.hash(elFork.ordinal ++ chainId ++ genesisRootHash)`.

Reply:
I think your approach is not going to be future proof because it doesn't include the data from the future fork configurations. We don't know what that data is going to be, so by introducing Fork ID with a `ConsensusConfig` as a field. If we're missing a ForkId computation for the future fork with say `QbftConsensusConfigV2`, it will fail in compile time. So by introducing this type we're introducing some additional type safety for the future extensions